api/resource/resource.go

```diff
@@ -296,20 +295,37 @@ func (r *Resource) MustYaml() string {
 	return string(yml)
 }
 
+func (r *Resource) getOptions() *types.GenArgs {
+	annotations := r.GetAnnotations()
+	if genOptsAnno, ok := annotations[utils.BuildAnnotationsGenOptions]; ok {
+		var genOpts types.GeneratorArgs
+		yaml.Unmarshal([]byte(genOptsAnno), &genOpts)
+		return types.NewGenArgs(&genOpts)
+	}
+	return &types.GenArgs{}
+}
+
 // SetOptions updates the generator options for the resource.
 func (r *Resource) SetOptions(o *types.GenArgs) {
-	r.options = o
+	if o == nil || o.Args == nil {
+		return
+	}
+	annotations := r.GetAnnotations()
+	b, _ := yaml.Marshal(o.Args)
```

Comment:
It can (and same comment below), but I'm curious why we would want to prefer one over the other?

Reply:
AFAIK kustomize _largely_ uses kyaml, which as you and I know very well uses yaml.v3 under the hood, which is a yaml 1.2 implementation. The sigs module is a wrapper around yaml.v2, which is a yaml 1.1 implementation. It's very unlikely that any differences would happen in marshalling a map of strings, and even more unlikely that it would matter within an ephemeral annotation like this. BUT in general I think we should stick to kyaml for consistency.

Reply:
done